src/features/applets-refresh/model/hooks/useRefreshResumeOnAppState.ts

```diff
@@ -0,0 +1,38 @@
+import { useRef, useEffect, Dispatch, SetStateAction } from 'react';
+import { AppState } from 'react-native';
+
+import { IS_ANDROID } from '@shared/lib';
+
+function useRefreshResumeOnAppState(
+  isRefreshing: boolean,
+  setIsRefreshing: Dispatch<SetStateAction<boolean>>,
+) {
+  const appState = useRef(AppState.currentState);
+
+  useEffect(() => {
+    if (IS_ANDROID) {
+      return;
+    }
+
+    const subscription = AppState.addEventListener('change', nextAppState => {
+      if (
+        appState.current.match(/inactive|background/) &&
+        nextAppState === 'active'
+      ) {
+        if (isRefreshing) {
+          setIsRefreshing(false);
+
+          setTimeout(() => {
+            setIsRefreshing(true);
+          }, 350);
```

Comment:
Please elaborate on why we need to wrap it in setTimeout and assign the exact delay of 350 ms

Reply:
I am wondering if we can utilize [flushSync](https://react.dev/reference/react-dom/flushSync). Could you please give it a try?

Reply:
Ah, gotcha :(